psp/models/recent_history.py

```diff
@@ -527,6 +527,33 @@ def _get_features(self, x: X, is_training: bool) -> Features:
         scalar_features["recent_power"] = 0.0 if recent_power_nan else recent_power
         scalar_features["recent_power_nan"] = recent_power_nan * 1.0
 
+        # recent power values
+        recent_power_values = data.sel(
+            ts=slice(x.ts - timedelta(minutes=recent_power_minutes), x.ts)
+        ).values
+
+        # make sure recent power values is the right length
+        if not hasattr(self, "_n_recent_power_values"):
+            self._n_recent_power_values = 0
+        if len(recent_power_values) < self._n_recent_power_values:
+            recent_power_values = np.pad(
+                recent_power_values,
+                (0, self._n_recent_power_values - len(recent_power_values)),
+                "constant",
+                constant_values=np.nan,
+            )
+        elif len(recent_power_values) > self._n_recent_power_values:
+            recent_power_values = recent_power_values[
+                len(recent_power_values) - self._n_recent_power_values :
+            ]
```

Comment:
Changes might need to be made when we use the flag --no-live-pv, to remove PV features during inference for metadata only backtests.

Reply:
yea, good shout, ill have a look at that. Probably the same for the `mean power values` above. 

Although the default is 0 values, so it might be ok

Reply:
Where is this `flag`?

Reply:
The new features would need to be removed from here: https://github.com/openclimatefix/pv-site-prediction/blob/7a46fd08a5b1358152c415e74b9174aa73f4ac5c/psp/models/multi.py#L30

Reply:
currently these pv features are hardcoded (in line 48):         
pv_derived_features = ["recent_power", "h_max", "h_median", "h_mean"]
nan_pv_derived_features = ["recent_power_nan", "h_max_nan", "h_median_nan", "h_mean_nan"]

Reply:
I think I found it